src/realm/sync/noinst/client_history_impl.cpp

```diff
@@ -1396,18 +1410,13 @@ void ClientHistoryImpl::fix_up_client_file_ident_in_stored_changesets(Transactio
 
     REALM_ASSERT(client_file_ident != 0);
     using Instruction = realm::sync::Instruction;
-
-    auto promote_global_key = [&](GlobalKey& oid) {
-        REALM_ASSERT(oid.hi() == 0); // client_file_ident == 0
-        oid = GlobalKey{uint64_t(client_file_ident), oid.lo()};
-    };
-
-    auto promote_primary_key = [&](Instruction::PrimaryKey& pk) {
-        mpark::visit(util::overload{[&](GlobalKey& key) {
-                                        promote_global_key(key);
-                                    },
-                                    [](auto&&) {}},
-                     pk);
+    bool did_modify;
```

Comment:
Initialize this to `false` rather than setting it explicitly below.

Reply:
Well, it should be set to false in each iteration. Anyway - I will change the implementation slightly.